drivers/timer/cavs_timer.c

```diff
@@ -19,35 +19,38 @@
  * all available CPU cores and provides a synchronized timer under SMP.
  */
 
-#define TIMER		0
-#define TIMER_IRQ	DSP_WCT_IRQ(TIMER)
+#define COMPARATOR_IDX  0 /* 0 or 1 */
+
+#define TIMER_IRQ	DSP_WCT_IRQ(COMPARATOR_IDX)
+
 #define CYC_PER_TICK	(CONFIG_SYS_CLOCK_HW_CYCLES_PER_SEC	\
 			/ CONFIG_SYS_CLOCK_TICKS_PER_SEC)
 #define MAX_CYC		0xFFFFFFFFUL
 #define MAX_TICKS	((MAX_CYC - CYC_PER_TICK) / CYC_PER_TICK)
 #define MIN_DELAY	(CONFIG_SYS_CLOCK_HW_CYCLES_PER_SEC / 100000)
 
 BUILD_ASSERT(MIN_DELAY < CYC_PER_TICK);
-BUILD_ASSERT(TIMER >= 0 && TIMER <= 1);
+BUILD_ASSERT(COMPARATOR_IDX >= 0 && COMPARATOR_IDX <= 1);
+
+#define WCTCS      (&CAVS_SHIM.dspwctcs)
```

Comment:
Why take the address of these here when most/all of the uses below then deref it?

Reply:
Clarity and style.  A macro being used like a pointer feels naturally like a MMIO register to me.  A macro that looks like data (and isn't otherwise used like memory, e.g. as a struct with fields as CAVS_SHIM is) seems weirder.

Reply:
I suppose a smart compiler would remove the *(&(x)) and replace with x as well potentially, so I think that makes sense, do you know if gcc or xcc do that today? Mostly curious at this point.